onnxruntime/core/providers/cpu/ml/tree_ensemble_common.h

```diff
@@ -0,0 +1,706 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+#include "tree_ensemble_aggregator.h"
+
+namespace onnxruntime {
+namespace ml {
+
+template <typename ITYPE, typename OTYPE>
+class TreeEnsembleCommon {
+ public:
+  int64_t n_targets_or_classes_;
+
+ protected:
+  std::vector<OTYPE> base_values_;
```

Comment:
All member variables should be private with get/set accessor as needed. 

[https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-private](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rc-private)
[https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rh-protected](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rh-protected)

Maybe this could be refactored so that these data parts are in a struct and a derived class could have a protected const getter to access that info.